src/PrestaShopBundle/DependencyInjection/Compiler/LoadServicesFromModulesPass.php

```diff
@@ -54,39 +59,29 @@ class LoadServicesFromModulesPass implements CompilerPassInterface
     public function __construct($containerName = '')
     {
         $this->configPath = '/config/' . (empty($containerName) ? '' : trim($containerName, '/') . '/');
+        $this->activeModulesPaths = (new ModuleRepository())->getActiveModulesPaths();
     }
 
     /**
      * {@inheritdoc}
      */
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasParameter('kernel.active_modules')) {
```

Comment:
Hi @PululuK 

Current way of retrieving active modules is using this parameter `kernel.active_modules`

ModuleRepository:: getActiveModulesPaths() logic seem different. It's based on the database content. So isn't this changing the behavior?


https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Module/Repository/ModuleRepository.php

Reply:
@matks see please https://github.com/PrestaShop/PrestaShop/blob/develop/app/AppKernel.php#L192

Reply:
Ha 😄 

But then isn't this PR duplicating the behavior? 🤔

Reply:
Can you please read the full code ?

Reply:
@matks  currently : 
- You use `getModulesPaths` and loop to check if exists in modules active


`ModuleRepository::getActiveModulesPaths` is equivalent, so more appropriate here I think.